agents/craftassist/craftassist_swarm_master.py

```diff
@@ -73,28 +87,150 @@ def init_controller(self):
         dialogue_object_classes["put_memory"] = PutMemoryHandler
         self.opts.block_data = craftassist_specs.get_block_data()
         self.opts.special_shape_functions = SPECIAL_SHAPE_FNS
-        # use swarm dialogue manager that does not track its own swarm behaviorss
+        low_level_interpreter_data = {
+            'block_data': craftassist_specs.get_block_data(),
+            'special_shape_functions': SPECIAL_SHAPE_FNS,
+            'color_bid_map': COLOR_BID_MAP,
+            'astar_search': astar,
+            'get_all_holes_fn': heuristic_perception.get_all_nearby_holes}
         self.dialogue_manager = SwarmDialogueManager(
             memory=self.memory,
             dialogue_object_classes=dialogue_object_classes,
             dialogue_object_mapper=DialogueObjectMapper,
             opts=self.opts,
+            low_level_interpreter_data=low_level_interpreter_data
         )
-    
+
+        self.handle_query_dict = {
+            "_db_read": self.memory._db_read,
+            "_db_read_one": self.memory._db_read_one,
+            "_db_write": self.memory._db_write,
+            "db_write": self.memory.db_write,
+            "tag": self.memory.tag,
+            "add_triple": self.memory.add_triple,
+            "check_memid_exists": self.memory.check_memid_exists,
+            "get_mem_by_id": self.memory.get_mem_by_id,
+            "basic_search": self.memory.basic_search
+        }
+
+    def task_step(self, sleep_time=0.25):
+        # TODO: add tag check to the query
+        query = "SELECT MEMORY FROM Task WHERE prio=-1"
+        _, task_mems = self.memory.basic_search(query)
+        for mem in task_mems:
+            if "swarm_worker_0" not in mem.get_tags():
```

Comment:
is this a check to see if this task belongs to the swarm master?  like is swarm_worker_0 the name of the master?  if yes, could we name/tag accordingly?

Reply:
yes, we could. In the latest pushed version. I removed the swarm_worker_0 tag but only tagging the swarm workers.